### PR TITLE
Fix test for gdc shared libs

### LINKED ISSUE #3952: gdc shared library test fails
The test case `test/D/SharedObjects/sconstest-gdc.py` fails if gdc is actually installed.  This is because it does a version check, certain old gdc versions did not support shared objects.

1. the `subprocess` call does not actually capture the output of the `gdc --version` call.
2. if it did capture it, it would still be wrong, because the wrong word of the output is examined: `version = result.decode().splitlines()[0].split()[3]`  (should be `[2]`).  Possibly a little too much magic in that line.

The offending file is `test/D/SharedObjects/Common/common.py`.  Should affect only tests.

### PR DESCRIPTION
**Pull request by mwichmann**
Fixes #3952

Test-only change, no docs (and per convention, no changelog)

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: test/D/SharedObjects/Common/common.py [hunk 1]
```diff
@@ -45,8 +45,10 @@ def testForTool(tool):
         test.skip_test("Required executable for tool '{0}' not found, skipping test.\n".format(tool))
 
     if tool == 'gdc':
-        result = check_output(('gdc', '--version'))
-        version = result.decode().splitlines()[0].split()[3]
+        cp = subprocess.run(('gdc', '--version'), capture_output=True)
+        # gdc (GCC) 11.1.1 20210531 (Red Hat 11.1.1-3)\nCopyright (C)...
```
Comment (bdbaddog) [inline test/D/SharedObjects/Common/common.py:51]: What did the older versions output look like? will this break those?

Comment (mwichmann) [reply test/D/SharedObjects/Common/common.py:51]: I don't have anything really old, but this is a really standard layout for the version message for the Gnu Compiler Collection: `tool (GCC) x.y.x reldate buiildinfo` so I'd be surprised if there's been any change for a very long time.

Comment (mwichmann) [reply test/D/SharedObjects/Common/common.py:51]: The gcc tool uses something also head-scratchy, and of course different:
```
    # --version variant:
    match = re.search(r'[0-9]+(\.[0-9]+)+', line)
    if match:
        version = match.group(0)
```

Dunno if that's better or worse

**Comment by bdbaddog** [comment]
Looks like this broke for pythons 3.5, 3.6, 3.8, 3.9

```
290/1213 (23.91%) /home/travis/virtualenv/python3.5.10/bin/python test/D/SharedObjects/sconstest-gdc.py
Traceback (most recent call last):
  File "test/D/SharedObjects/sconstest-gdc.py", line 31, in <module>
    testForTool('gdc')
  File "/home/travis/build/SCons/scons/test/D/SharedObjects/Common/common.py", line 48, in testForTool
    cp = subprocess.run(('gdc', '--version'), capture_output=True)
  File "/opt/python/3.5.10/lib/python3.5/subprocess.py", line 383, in run
    with Popen(*popenargs, **kwargs) as process:
TypeError: __init__() got an unexpected keyword argument 'capture_output'

```

**Comment by mwichmann** [comment]
Sigh, one part is "too new" Python construct, the others because apparently the version string *isn't* laid out the same way with the build we get on the Travis instances.  There it looks llike (well, based on a copy I have here):
```
gdc (Ubuntu 10.2.0-5ubuntu1~20.04) 10.2.0
```
So I guess we need to use a regex after all.

**Comment by mwichmann** [comment]
Should be fixed now,.

**Pull request merged**